### PR TITLE
[Fix](nereids) Support insert ignore statement

### PR DESCRIPTION
**Pull request by bobhan1**
## Proposed changes

a fix for https://github.com/apache/doris/pull/21773

## Further comments

If this is a relatively large or complex change, kick off the discussion at [dev@doris.apache.org](mailto:dev@doris.apache.org) by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

**Comment by bobhan1** [comment]
run buildall

File: fe/fe-core/src/main/java/org/apache/doris/nereids/glue/translator/PhysicalPlanTranslator.java [hunk 1]
```diff
@@ -353,6 +354,19 @@ public PlanFragment visitPhysicalOlapTableSink(PhysicalOlapTableSink<? extends P
...
+            OlapTable dstTable = olapTableSink.getTargetTable();
+            if (dstTable.getKeysType() != KeysType.UNIQUE_KEYS || !dstTable.getEnableUniqueKeyMergeOnWrite()) {
+                throw new AnalysisException("ignore mode can only be enabled if the target table is "
+                        + "a unique table with merge-on-write enabled.");
+            } else if (olapTableSink.isPartialUpdate()) {
+                throw new AnalysisException("ignore mode can't be used in partial update.");
+            } else if (olapTableSink.getTargetTable().hasSequenceCol()) {
+                throw new AnalysisException("ignore mode can't be used if the target table has sequence column, "
+                        + "but table[" + dstTable.getName() + "] has sequnce column.");
+            }
```
Comment (morrySnow) [inline fe/fe-core/src/main/java/org/apache/doris/nereids/glue/translator/PhysicalPlanTranslator.java:367]: i think we should check this in bindSink

**Comment by bobhan1** [comment]
run buildall

**Review by zhannngchen — APPROVED** [review]
LGTM

**Comment by bobhan1** [comment]
The feature is too complicate too handle, so we decide not to support it now.

**Pull request closed without merging**